src/tests/components/JSONOutput.test.js

```diff
@@ -15,62 +24,65 @@ afterEach(() => {
   container = null;
 });
 
-it('Renders with the default heading if displaySUSHI is false', () => {
+it('Renders with the default text if displaySUSHI is false, isObject is false, and no text', () => {
   // Initial case, nothing from SUSHI is displayed
-  const { getByText, queryByText } = render(
-    <JSONOutput displaySUSHI={false} text={'Hello World'} errorsAndWarnings={[]} />,
+  const { getByText } = render(
+    <JSONOutput displaySUSHI={false} isObject={false} text={''} errorsAndWarnings={[]} />,
     container
   );
-  const headerElement = getByText(/SUSHI Output/i);
-  const resultsElement = queryByText(/Results/i);
-  const errorsElement = queryByText(/Errors/i);
 
-  expect(headerElement).toBeInTheDocument();
-  expect(resultsElement).not.toBeInTheDocument();
-  expect(errorsElement).not.toBeInTheDocument();
-});
-
-it('Renders with the proper heading and updates with proper text when not an object', () => {
-  const { getByText, queryByText } = render(
-    <JSONOutput displaySUSHI={true} text={'Hello World'} isObject={false} errorsAndWarnings={[]} />,
-    container
-  );
-  const resultsElement = queryByText(/Results/i);
-  const errorsElement = queryByText(/Errors/i);
-  const textElement = getByText(/Hello World/i);
+  // Because the editor is in JSON mode, the text is split up differently than the fsh mode
+  const placeholderText = getByText('Edit and view FHIR Definitions here!');
 
-  expect(resultsElement).not.toBeInTheDocument();
-  expect(errorsElement).not.toBeInTheDocument();
-  expect(textElement).toBeInTheDocument(); // Mimics the 'loading...' case
+  expect(placeholderText).toBeInTheDocument();
 });
 
-it('Renders with the proper headings when text is an object (SUSHI Package)', () => {
-  const { getByText, queryByText } = render(
-    <JSONOutput displaySUSHI={true} text={JSON.stringify({ profiles: [] })} isObject={true} errorsAndWarnings={[]} />,
+it('Renders with the proper text and updates with proper text when not an object', () => {
+  const { getByText } = render(
+    <JSONOutput displaySUSHI={true} text={'Loading...'} isObject={false} errorsAndWarnings={[]} isWaiting={true} />,
     container
   );
-  const resultsElement = getByText(/Results/i);
-  const errorsElement = queryByText(/Errors/i);
+  const textElement = getByText(
+    (content, element) => element.tagName.toLowerCase() === 'span' && content.startsWith('Loading')
+  );
 
-  expect(resultsElement).toBeInTheDocument(); // isObject is true so results are printed
-  expect(errorsElement).not.toBeInTheDocument(); // No errors
+  expect(textElement).toBeInTheDocument(); // Mimics the 'loading...' case
+  expect(textElement.parentNode.textContent).toEqual('Loading...');
 });
 
-it('Renders error messages if present', () => {
+// TODO: CodeMirrorComponent doesn't get the package text properly - not sure why
+it.skip('Renders with the first profile when text is an object (SUSHI Package)', async () => {
```

Comment:
I played around with this skipped test case for a little bit and was stumped on it too. It's really not making sense to me that the `getByText` function is not finding the content there but working just fine in the test right above it 😕. We can debug this later though and everything is working great, so I'm still gonna go ahead and approve. This feature is looking really awesome so far, nice work!

Reply:
Yeah, the weird thing to me is that you can see the output when the test fails and it looks like there's only a `}` in the editor, which is really odd. As long as you think we're okay holding off on this test for now, I'm good! I'd love some help eventually sorting out the test when we do fix it.